mmcv/ops/csrc/common/cuda/iou3d_cuda_kernel.cuh

```diff
@@ -216,21 +216,21 @@ __device__ inline float iou_bev(const float *box_a, const float *box_b) {
   return s_overlap / fmaxf(sa + sb - s_overlap, EPS);
 }
 
-__global__ void iou3d_boxes_iou3d_forward_cuda_kernel(const int num_a,
-                                                      const float *boxes_a,
-                                                      const int num_b,
-                                                      const float *boxes_b,
-                                                      float *ans_iou) {
-  CUDA_2D_KERNEL_LOOP(b_idx, num_b, a_idx, num_a) {
-    if (a_idx >= num_a || b_idx >= num_b) {
-      return;
-    }
-
-    const float *cur_box_a = boxes_a + a_idx * 7;
-    const float *cur_box_b = boxes_b + b_idx * 7;
-    float cur_iou_bev = iou_bev(cur_box_a, cur_box_b);
-    ans_iou[a_idx * num_b + b_idx] = cur_iou_bev;
+__global__ void iou3d_boxes_overlap_bev_forward_cuda_kernel(
+    const int num_a, const float *boxes_a, const int num_b,
+    const float *boxes_b, float *ans_overlap) {
+  // params boxes_a: (N, 7) [x, y, z, dx, dy, dz, heading]
+  // params boxes_b: (M, 7) [x, y, z, dx, dy, dz, heading]
+  const int a_idx = blockIdx.y * THREADS_PER_BLOCK + threadIdx.y;
```

Comment:
The block you used to launch the kernel is:
```c++
  dim3 blocks(GET_BLOCKS(num_b, THREADS_PER_BLOCK_IOU3D),
              GET_BLOCKS(num_a, THREADS_PER_BLOCK_IOU3D));
```

`THREADS_PER_BLOCK` is not safe here.